app/script/location/LocationRepository.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.location = window.z.location || {};
+
+z.location.LocationRepository = class LocationRepository {
+  constructor(locationService) {
+    this.getMapsUrl = locationService.getMapsUrl.bind(locationService);
+    this.getLocation = locationService.getLocation.bind(locationService);
+  }
```

Comment:
Let's bind them in the location service so that if we are covered if ever call them from elsewhere

Reply:
I would argue that this is the responsibility of the consumer to bind the function to the right object.
The class should not be aware how it's methods are going to be used.

Reply:
Let's go with 
```
this.locationService = locationService;
```
and just call the methods directly from the service where they are needed. So it is clear where these methods belong and which context they are using without the need to awkwardly bind them anywhere.